demo/site/src/util/createGraphQLClient.ts

```diff
@@ -1,10 +1,8 @@
+import { PreviewData } from "@src/pages/api/preview";
 import { GraphQLClient } from "graphql-request";
 
-interface GraphQLClientOptions {
-    includeInvisiblePages: boolean;
-    includeInvisibleBlocks: boolean;
-    previewDamUrls: boolean;
-}
+type GraphQLClientOptions = PreviewData;
```

Comment:
maybe you could swap location where the type is defined:

keep GraphQLClientOptions and let api/preview use it. Seems more logical than a dependency of the graphql client to the preview api